src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs

```diff
@@ -166,6 +165,38 @@ public virtual async Task<OrderDetailViewModel> GetOrderDetailViewModelForGuestA
             return viewModel;
         }
 
+        /// <summary>
+        /// Update Order Customer Id
+        /// </summary>
+        /// <param name="param"></param>
+        /// <returns></returns>
+        public virtual async Task<Overture.ServiceModel.Orders.Order> UpdateOrderCustomerAsync(UpdateOrderCustomerParam param)
+        {
+            if (param == null) { throw new ArgumentNullException(nameof(param)); }
+            if (param.CultureInfo == null) { throw new ArgumentException(nameof(param.CultureInfo)); }
+            if (string.IsNullOrWhiteSpace(param.Scope)) { throw new ArgumentException(nameof(param.Scope)); }
+            if (string.IsNullOrWhiteSpace(param.OrderNumber)) { throw new ArgumentException(nameof(param.OrderNumber)); }
+            if (param.CustomerId == default) { throw new ArgumentException(nameof(param.CustomerId)); }
+
+            var order = await OrderRepository.GetOrderAsync(param).ConfigureAwait(false);
+            var customer = await OrderRepository.GetCustomerByIdAsync(param.CustomerId, param.Scope).ConfigureAwait(false);
+
+            if (order == null || customer == null || order.Cart.Customer.Email != customer.Email)
+            {
+                return null;
+            }
```

Comment:
I think it is better to check whether the order is null right after this request, not to send the second request for var customer variable in case var order is null